samples/bluetooth/nfc_pairing/src/main.c

```diff
@@ -0,0 +1,477 @@
+{
+	advertising_continue();
+}
+
+static void auth_cancel(struct bt_conn *conn)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	printk("Pairing cancelled: %s\n", addr);
+}
+
+static void pairing_confirm(struct bt_conn *conn)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	bt_conn_auth_pairing_confirm(conn);
+
+	printk("Pairing confirmed: %s\n", addr);
+}
+
+static void lesc_obb_data_set(struct bt_conn *conn,
+			      struct bt_conn_oob_info *info)
+{
+	int err;
+
+	if (info->lesc.oob_config != BT_CONN_OOB_LOCAL_ONLY) {
+		printk("LESC OOB config not supported\n");
+		return;
+	}
+
+	/* Pass only local OOB data. */
+	err = bt_le_oob_set_sc_data(conn, &oob_local.le_sc_data, NULL);
+	if (err) {
+		printk("Error while setting OOB data: %d\n", err);
+	} else {
+		printk("Successfully provided LESC OOB data\n");
+	}
+}
+
+static void legacy_tk_value_set(struct bt_conn *conn)
+{
+	int err;
+
+	err = bt_le_oob_set_legacy_tk(conn, tk_value);
+	if (err) {
+		printk("TK value set error: %d\n", err);
+	}
+}
+
+static void auth_oob_data_request(struct bt_conn *conn,
+				  struct bt_conn_oob_info *info)
+{
+	if (info->type == BT_CONN_OOB_LE_SC) {
+		printk("LESC OOB data requested\n");
+		lesc_obb_data_set(conn, info);
+	}
+
+	if (info->type == BT_CONN_OOB_LE_LEGACY) {
+		printk("Legacy TK value requested\n");
+		legacy_tk_value_set(conn);
+	}
+}
+
+static void pairing_complete(struct bt_conn *conn, bool bonded)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	printk("Pairing completed: %s, bonded: %d\n", addr, bonded);
+
+	k_work_submit(&pair_msg_work);
+}
+
+static void pairing_failed(struct bt_conn *conn, enum bt_security_err reason)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	printk("Pairing failed conn: %s, reason %d\n", addr, reason);
+
+	k_work_submit(&pair_msg_work);
+}
+
+static enum bt_security_err pairing_accept(struct bt_conn *conn,
+				const struct bt_conn_pairing_feat *const feat)
+{
+	if (feat->oob_data_flag && (!(feat->auth_req & AUTH_SC_FLAG))) {
+		bt_set_oob_data_flag(true);
+	}
+
+	return BT_SECURITY_ERR_SUCCESS;
+
+}
+
+static struct bt_conn_auth_cb conn_auth_callbacks = {
+	.cancel = auth_cancel,
+	.pairing_confirm = pairing_confirm,
+	.oob_data_request = auth_oob_data_request,
+	.pairing_complete = pairing_complete,
+	.pairing_failed = pairing_failed,
+	.pairing_accept = pairing_accept,
+};
+
+static void connected(struct bt_conn *conn, u8_t err)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	if (err) {
+		if (err == BT_HCI_ERR_ADV_TIMEOUT) {
+			printk("Direct advertising to %s timed out\n", addr);
+			k_work_submit(&adv_work);
+		} else {
+			printk("Failed to connect to %s (%u)\n", addr, err);
+		}
+
+		return;
+	}
+
+	conn_cnt++;
```

Comment:
shouldn't we call `bt_conn_ref()` here?

Reply:
No, we want to count connection not refering existing one

Reply:
OK, makes sense